packages/nx/src/adapter/ngcli-adapter.ts

```diff
@@ -51,10 +53,73 @@ import {
   CustomHasher,
   Executor,
   ExecutorConfig,
+  ExecutorContext,
   ExecutorsJson,
   TaskGraphExecutor,
 } from '../config/misc-interfaces';
 import { readPluginPackageJson } from '../utils/nx-plugin';
+import { WorkspaceNodeModulesArchitectHost as AngularWorkspaceNodeModulesArchitectHost } from '@angular-devkit/architect/node';
+
+export async function createBuilderContext(
+  builderInfo: {
+    builderName: string;
+    description: string;
+    optionSchema: any;
+  },
+  context: ExecutorContext
+) {
+  require('nx/src/adapter/compat');
+  const fsHost = new NxScopedHost(context.root);
+  const { workspace } = await workspaces.readWorkspace(
+    'angular.json',
+    workspaces.createWorkspaceHost(fsHost)
+  );
+  const architectHost = new WrappedWorkspaceNodeModulesArchitectHost(
+    workspace,
+    context.root
+  );
+
+  const registry = new schema.CoreSchemaRegistry();
+  registry.addPostTransform(schema.transforms.addUndefinedDefaults);
+  registry.addSmartDefaultProvider('unparsed', () => {
+    // This happens when context.scheduleTarget is used to run a target using nx:run-commands
+    return [];
+  });
+  const { Architect } = require('@angular-devkit/architect');
+
+  const architect: Architect = new Architect(architectHost, registry);
+
+  const builderContext: import('@angular-devkit/architect').BuilderContext = {
+    workspaceRoot: context.root,
+    target: {
+      project: context.projectName,
+      target: context.targetName,
+      configuration: context.configurationName,
+    },
+    builder: {
+      ...builderInfo,
+    },
+    logger: getLogger(),
+    id: 1,
+    currentDirectory: process.cwd(),
+    scheduleTarget: architect.scheduleTarget,
+    addTeardown(teardown: () => Promise<void> | void) {},
+    validateOptions: architect['_scheduler'].validateOptions,
```

Comment:
I couldn't find `validateOptions` in the `SimpleScheduler` which is the type of the `_scheduler` (https://github.com/angular/angular-cli/blob/333da08ea29e466dd883dfd4742088deb73b25d0/packages/angular_devkit/architect/src/architect.ts#L382). The `validateOptions` in the builder context is implemented here https://github.com/angular/angular-cli/blob/333da08ea29e466dd883dfd4742088deb73b25d0/packages/angular_devkit/architect/src/create-builder.ts#L166-L176. It schedules a job named `..validateOptions`.

Unless I'm missing something, we're effectively setting it to `undefined` which is probably not what we want. The `validateOptions` in the context is used internally by some builders to validate the options of other targets they execute (e.g. builders with an option like `browserTarget`).

From what I saw, this is not an issue with the current `browser-esbuild` builder because it's not using that function, but this function is a generic utility and can be used with other builders.

Reply:
Something to look into, but as it is not used by esbuild browser, it could be something that is addressed in a follow up PR.

Reply:
Even if not used, providing here something that it just doesn't exist is wrong. If we don't plan to support it now and add the support later, we should add least be explicit about it adding a noop function with a TODO note.